Add new tcr arena

Adimote opened the pull request:
Add the arena for the Tin Can Rally 2018 design

Adimote committed:
copy tcr2017 into new files

Adimote committed:
copy tcr2017 into new files

Adimote committed:
change default arena to tcr2018

Adimote committed:
change default arena to tcr2018

Adimote committed:
start on new arena layout

Adimote committed:
start on new arena layout

Adimote committed:
finish TCR 2018 arena

Adimote committed:
finish TCR 2018 arena

Adimote committed:
s/2017/2018 in header comment

Adimote committed:
s/2017/2018 in header comment

Adimote committed:
whitespace and remove id for tin cans

Adimote committed:
whitespace and remove id for tin cans

Adimote committed:
remove pointless enumerate()

Adimote committed:
remove pointless enumerate()

Adimote committed:
Fix edge case where robot wont grab

The Robot wouldn't grab if the token was above the robot while the
marker was below, this is because the bearings of the robot and
token would read -pi and pi. The offset would then read as 2pi,
instead of 0 and therefore the token wouldn't be grabbed.

This is fixed by adding an offset to wrap the total offset to
never be over pi.

Adimote committed:
Fix edge case where robot wont grab

The Robot wouldn't grab if the token was above the robot while the
marker was below, this is because the bearings of the robot and
token would read -pi and pi. The offset would then read as 2pi,
instead of 0 and therefore the token wouldn't be grabbed.

This is fixed by adding an offset to wrap the total offset to
never be over pi.

PeterJCLaw commented on sb/robot/arenas/tin_can_rally_2018.py:
```diff
@@ -0,0 +1,244 @@
+            (3.4 - 4, 6.0 - 4),
+        ]
+
+        token_locations += [TCRArena2018.rotate(pos, pi) for pos in token_locations]
```
Minor: maybe call this as `self.rotate` rather than `TCRArena2018.rotate`? While either works, it's not really important here that it's a class method, though including that makes it more verbose (and had me wondering which class I was currently reading).

Adimote replied:
fixed in 7fa77bb01ce95c89cb13790d64eca5638f2aa90b

kierdavis commented:
Why is this set to be merged into `pirate-islands-fixup` rather than `master`?

PeterJCLaw reviewed (commented):
Not sure how much of this is this PR vs the baseline, though here are some thoughts:
- the cans aren't very easy to see
- the low obstacles in the shortcut are the same as the larger blocks, it would be good to make them distinct
- neither the obstacles or the staging blocks are obvious what they are as they have no fill
- the sourcebots logo is very prominent and makes it hard to see what else is in the arena

PeterJCLaw commented:
@kierdavis `pirate-islands-fixup` is the branch used last year and which contains the changes to move to the SourceBots api, however it wasn't ever merged into `master`. By targetting that branch, rather than `master` we get a nicer diff on github.

Adimote committed:
make central walls solid and tweak positions

Adimote committed:
make central walls solid and tweak positions

Adimote committed:
fix can colour, recolour immobile obstacles

Adimote committed:
fix can colour, recolour immobile obstacles

Adimote committed:
move logo to the bottom of the screen

Adimote committed:
move logo to the bottom of the screen

Adimote committed:
nudge an obstacle to be more central

Adimote committed:
nudge an obstacle to be more central

Adimote committed:
nudge a boundary line to not have a gap

Adimote committed:
nudge a boundary line to not have a gap

Adimote committed:
shrink the robot so it actually fits through obstacles

Adimote committed:
shrink the robot so it actually fits through obstacles

Adimote committed:
remove unneeded class ref

Adimote committed:
remove unneeded class ref

Adimote commented:
@PeterJCLaw I've made a stab at fixing all of your issues, please re-review

PeterJCLaw commented:
@Adimote thanks, the visuals look good now. Unfortunately the robot doesn't seem to be able to move while carrying a can, though it can push them about which is good.
I had been expecting that the staging blocks would be touching in the middle, though I don't know whether that interferes with the super can?

PeterJCLaw committed:
Restore the test.py from master

Which doesn't use vision.

PeterJCLaw committed:
Adapt test.py to the SourceBots API

PeterJCLaw committed:
Hide the vision related attributes

Deliberately doesn't remove the related code so that this is easy
to restore (we're only removing it for use with a specific arena).

Adimote committed:
fix robot not rotating when can is grabbed

Adimote committed:
fix robot not rotating when can is grabbed

Adimote committed:
fix robot not rotating when can is grabbed

Adimote committed:
Merge branch 'add-new-tcr-arena' of github.com:sourcebots/simulator into add-new-tcr-arena

Adimote committed:
add platform under gold can to imply it's above the walls

Adimote commented:
@PeterJCLaw I've fixed robots rotating with cans now, and yes I had to nudge the staging blocks apart a bit to ensure the gold token was still grabbable. I've now added the platform under the can to hide that it's actually in a gap.

PeterJCLaw committed:
Merge branch 'no-vision' into add-new-tcr-arena

Adimote commented:
I've now merged this into the `smallpeice-2018-tin-can-rally` branch
